yarn-project/aztec-rpc/src/synchroniser/synchroniser.ts

```diff
@@ -172,7 +229,15 @@ export class Synchroniser {
     if (processor) {
       return;
     }
-    this.noteProcessors.push(new NoteProcessor(publicKey, keyStore, this.db, this.node));
+
+    const noteProcessor = new NoteProcessor(publicKey, keyStore, this.db, this.node);
+    if (this.synchedToBlock === 0) {
+      // The main sync thread was never started before and for this reason the synchroniser does not have to catch up
+      this.noteProcessors.push(noteProcessor);
+    } else {
+      // The main sync thread was started before and for this reason the synchroniser has to catch up
+      this.noteProcessorsToCatchUp.push(noteProcessor);
+    }
```

Comment:
@benesjan sorry for going back to a merged PR, but I think you may have a race condition here: if a `noteProcessor` is added on the first run of `work` during `await this.updateBlockInfoInBlockTxs(blockContexts)`, which happens after the note processors are notified but before updating `synchedToBlock`, then it will be pushed to `noteProcessors` but will never get the notes from the first iteration.

If the above is correct, maybe you can solve it with a flag `isStarted` that you set as soon as `start` is called, and check that instead of `synchedToBlock`?

Reply:
I think @spalladino is right. Perhaps a clean solution is to always add the note processor to `noteProcessorsToCatchUp` and have `workNoteProcessorCatchUp` transfer it to `noteProcessors` in all scenarios.

Reply:
@spalladino great catch! Will address